### PR TITLE
[DTP-951] Implement `LiveObjects.getRoot()` method

### PR DESCRIPTION
**Pull request by VeskeR**
This PR is based on https://github.com/ably/ably-js/pull/1890, please review it first.

Tests for SYNC sequence are added in https://github.com/ably/ably-js/pull/1894.

Resolves [DTP-951](https://ably.atlassian.net/browse/DTP-951)

[DTP-951]: https://ably.atlassian.net/browse/DTP-951?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

- **New Features**
	- Introduced event-driven programming capabilities with the addition of `EventEmitter` and `SyncCompleted` event in the `LiveObjects` class, enhancing synchronization handling.
  
- **Bug Fixes**
	- Improved test suite for `LiveObjects` to ensure proper channel attachment before executing tests, leading to more reliable test outcomes.

### CODE REVIEW

**Comment by VeskeR** [comment]
@coderabbitai review

**Review by mschristensen — APPROVED** [review]
Looks good to me

File: src/plugins/liveobjects/liveobjects.ts [hunk 1]
```diff
@@ -20,14 +28,24 @@ export class LiveObjects {
...
+      // SYNC is finished, can return immediately root object from pool
+      return this._liveObjectsPool.get(ROOT_OBJECT_ID) as LiveMap;
+    }
+
+    // otherwise wait for SYNC sequence to finish
+    return new Promise((res) => {
+      this._eventEmitter.once(LiveObjectsEvents.SyncCompleted, () => {
+        res(this._liveObjectsPool.get(ROOT_OBJECT_ID) as LiveMap);
+      });
+    });
```
Comment (owenpearson) [inline src/plugins/liveobjects/liveobjects.ts:48]:
```suggestion
    // SYNC is currently in progress, wait for SYNC sequence to finish
    if (this._syncInProgress) {
      await this._eventEmitter.once(LiveObjectsEvents.SyncCompleted)
    }
    
    return this._liveObjectsPool.get(ROOT_OBJECT_ID) as LiveMap;
```

^ much simpler + avoids mixing promises + async/await

**Review by owenpearson — COMMENTED** [review]
looks good, should make this change though:

File: src/plugins/liveobjects/liveobjects.ts (hunk 1 above)
Comment (VeskeR) [reply src/plugins/liveobjects/liveobjects.ts]: oh cool, didn't realize that there is an `.once` overload that returns a Promise. accepted the change

**Review by owenpearson — APPROVED** [review]

**Pull request merged**